sdk/include/platform/sunburst/platform-gpio.hh

```diff
@@ -138,6 +292,153 @@ struct SonataGPIO
 	 */
 	SonataJoystick read_joystick() volatile
 	{
-		return static_cast<SonataJoystick>(input & 0x1f);
+		return static_cast<SonataJoystick>(input & Inputs::Joystick);
+	}
+};
+
+/**
+ * A driver for Sonata's Raspberry Pi HAT Header GPIO.
+ *
+ * Documentation source:
+ * https://lowrisc.org/sonata-system/doc/ip/gpio.html
+ */
+struct SonataGpioRaspberryPiHat : SonataGpioBase
+{
+	/**
+	 * The mask of bits of the `output` register that contain meaningful GPIO
+	 * output.
+	 */
+	static constexpr uint32_t output_mask()
```

Comment:
I think this doesn't do what you think.  You probably want to either make this virtual (not ideal) or use CRTP so that the superclass can use it.